woocommerce/woocommerce
# Add custom image in downloads editor

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR serves as a follow-up to [PR 40835](https://github.com/woocommerce/woocommerce/pull/40835).  It adds a custom placeholder for downloadable products.

Design
AkVNGImLgSqCObTQ3idVn7-fi-6970_265448

Closes #41254.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor and the virtual-downloadable products (`product-virtual-downloadable`) using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New**
3. In the `General` tab, scroll to the bottom of the page where a new `Downloads` section should appear.
4. By default, the list displays files with their actual names, including the extension/format, after uploading one or more files. Each file is listed as a separate item with options to remove (using `X`) and `Edit` them.
6. Click the `Edit` button. A modal will be visible.
7. If the uploaded file was an image, confirm that the modal displays it.
8. Try uploading another file (not an image).
9. Verify that the custom image is visible next to the file name.
10. Verify that the `Replace` button works as expected.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
